docs/development_suite/set-up-infrastructure/enhanced-project-workflow.md

```diff
@@ -0,0 +1,136 @@
+---
+id: enhanced-project-workflow
+title: Enhanced Project Workflow
+sidebar_label: Enhanced Project Workflow
+---
+
+:::caution
+This is a **BETA** feature and, as such, is currently under active development. Pay attention using it.
+:::
+
+The **Enhanced Project Workflow** feature aims to improve the developer experience when working with the Console, as well as pave the way for the adoption of previously unavailable opportunities.
+
+## Changes to the Console workflow
+
+The following sections will highlight the changes made to the established Console workflow, regarding in particular the new management of the project logical state, and the new deployment workflow.
+
+### New configuration management strategy
+
+The most relevant change in the Console’s behavior is that the project’s logical state will not be saved on a Git provider anymore. This will result in significant **performance improvement** during the main Console workflows, since contacting the Git provider implied a considerable overhead that will now be reduced.
+
+This change also means that the Console will not rely on the typical Git-based instruments anymore and will instead adopt a **new configuration management system**, while still maintaining consistency with the Console’s established workflows.
+
+With the new workflow, branches and tags become **revisions** and **versions**, underscoring that the information associated with them is not stored within a Git repository anymore. The concept of commit is not present anymore and the commit history is replaced by a **version history**: the user can create versions, similar to Git tags, to keep track of the project’s evolution.
+
+### New deployment workflow
+Another significant change is the way Console configurations are saved and deployed: Kubernetes **configurations will be generated and committed** to the Git repository only **during the deployment process**. 
+
+Before deploying the configurations to the cluster, users can **review** which workload changes are going to be deployed by opening the review modal. This operation gives the developer a clear view of what is going to be released on the runtime environment, and it is mandatory if the user is deploying on a production environment. 
+
+Moreover, in the new workflow the user will be able to **export Kubernetes configurations** by clicking on a download button.
+:::note  
+The review and export features are still in development and will be available in future releases.
+:::
+
+:::caution
+When deploying your configurations to the cluster, keep in mind that only the changes pushed to the repository’s default branch will be considered.
```

Comment:
Do we want to specify this?

```suggestion
When deploying your configurations to the cluster, keep in mind that only the changes pushed to the repository’s default branch will be considered.

You can change the repository branch using by changing the `project.repository.configurationBranch` property 
```

Reply:
Is this a property manageable in some way at back-office level? Or in some other way?

Reply:
As of now you have to change it from the backoffice, we can consider moving it into project settings in the repository tab